tests/s3/copy_object/test_parallel_copy_object.py

```diff
@@ -365,3 +381,140 @@ def test_44793(self):
+                                            numproc=3)
+            dest_resp = self.s3_obj.object_info("dest-bucket1", f"{obj_type}-obj2")
+            src_resp = self.s3_obj.object_info("src-bucket1", f"{obj_type}-obj3")
+            assert_utils.assert_equal(dest_resp[1]["ETag"], src_resp[1]["ETag"])
+            src_resp = self.s3_obj.object_info("src-bucket1", f"{obj_type}-obj4")
+            assert_utils.assert_equal(dest_resp[1]["ETag"], src_resp[1]["ETag"])
+            dest_resp1 = self.s3_obj.object_info("dest-bucket2", f"{obj_type}-obj4")
+            assert_utils.assert_equal(dest_resp[1]["ETag"], src_resp[1]["ETag"])
+        LOGGER.info("ENDED: Test Reverse copy operations (simple and multipart source objects)")
+
+    @pytest.mark.s3_ops
+    @pytest.mark.s3_object_copy
+    @pytest.mark.tags("TEST-44801")
+    def test_44801(self):
+        """Test Cross source and destination bucket copy"""
+        LOGGER.info("STARTED: Test Cross source and destination bucket copy"
+                    "(simple and multipart source objects)")
+        for obj_type in ["simple", "multipart"]:
+            if obj_type == "simple":
+                resp1 = self.create_put_object("src-bucket1", f"{obj_type}-obj1", 10)
+                resp2 = self.create_put_object("src-bucket1", f"{obj_type}-obj4", 1000)
+            else:
+                resp1 = self.s3mp_obj.simple_multipart_upload("src-bucket1", f"{obj_type}-obj1",
+                                                              512, self.file_path, 4)
+                resp2 = self.s3mp_obj.simple_multipart_upload("src-bucket1", f"{obj_type}-obj4",
+                                                              2048, self.file_path, 4)
+            LOGGER.info("Uploaded %s-obj1 to src-bucket1", obj_type)
+            LOGGER.info("Uploaded %s-obj1 to src-bucket4", obj_type)
+            self.multi_parallel_copy_object(("src-bucket1", f"{obj_type}-obj1", "dest-bucket1",
+                                            f"{obj_type}-obj2"),
+                                            ("src-bucket1", f"{obj_type}-obj1", "dest-bucket1",
+                                            f"{obj_type}-obj3"),
+                                            ("src-bucket1", f"{obj_type}-obj4", "dest-bucket1",
+                                            f"{obj_type}-obj2"),
+                                            ("src-bucket1", f"{obj_type}-obj4", "dest-bucket1",
+                                            f"{obj_type}-obj3"),
+                                            numproc=4)
+            resp = self.s3_obj.object_info("dest-bucket1", f"{obj_type}-obj2")
+            assert_utils.assert_equal(resp2[1]["ETag"], resp[1]["ETag"])
+            resp = self.s3_obj.object_info("dest-bucket1", f"{obj_type}-obj3")
+            assert_utils.assert_equal(resp2[1]["ETag"], resp[1]["ETag"])
+        LOGGER.info("ENDED: Test Cross source and destination bucket copy"
```

Comment:
last step of listing contents of bucket is not implemented here

Reply:
in a source and destination bucket , we need to check which object are present and are they valid as we created/copied to specific bucket . I think Ketan already implemented it in last sprint, you can reuse it